core/services/workflows/syncer/workflow_registry.go

```diff
@@ -60,6 +55,30 @@ type GetWorkflowMetadata struct {
 	SecretsURL   string
 }
 
+type SyncStrategy string
+
+const (
+	SyncStrategyEvent          = "event"
+	SyncStrategyReconciliation = "reconciliation"
+	defaultSyncStrategy        = SyncStrategyEvent
+)
+
+func (s *SyncStrategy) Validate() error {
```

Comment:
Just make this value receiver; it should still work

Reply:
Small stylistic comment: I prefer a more procedural style that puts logic inside service objects (i.e. the registry syncer) over spreading it around in various OO-style objects. IMO this reads easier (the logic is right there and I don't need to go see what validate does). Only if the object is being used in multiple places would that trigger a refactor into some commonly shared methods.

(This isn't a direction to change in that way; I think this is more a style/preference point).